src/main/java/user_code_resolver/UserCodeResolver.java

```diff
@@ -0,0 +1,222 @@
+package user_code_resolver;
+
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.javers.core.Javers;
+import org.javers.core.JaversBuilder;
+import org.javers.core.diff.Change;
+import org.javers.core.diff.Diff;
+import org.javers.core.diff.ListCompareAlgorithm;
+import org.javers.core.diff.changetype.container.ContainerElementChange;
+import org.javers.core.diff.changetype.container.ElementValueChange;
+import org.javers.core.diff.changetype.container.ListChange;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Stream;
+
+/**
+ * The type User code resolver.
+ */
+public class UserCodeResolver {
+
+    private File file = new File("userCode.json");
+
+    /**
+     * Instantiates a new User code resolver.
+     *
+     * @param folder the folder
+     * @throws IOException the io exception
+     */
+    public UserCodeResolver(File folder) throws IOException {
+        Project project;
+        ObjectMapper objectMapper = new ObjectMapper();
+        if (this.file != null && this.file.length() > 0) {
+            project = objectMapper.readValue(this.file, Project.class);
```

Comment:
Ich bin nicht 100% mit der Funktionsweise vom ObjectMapper vertraut, aber brauchst du hier die Klasse `Project` wirklich? Im Prinzip ist alles was in der Klasse steckt ja nur eine Liste mit `UserFiles`. Deswegen würde ich vermuten, dass man im `UserCodeResolver` auch einfach eine Variable `List<UserFiles>` erstellen kann. Wenn der ObjectMapper aber `Project.class` zwingend braucht, dann können wir das auch so lassen. Würde es aber trotzdem mal prüfen.